modules/framework/jvm/src/main/scala/SbtTask.scala

```diff
@@ -30,7 +31,7 @@ private[framework] class SbtTask(
     loggerPermit.acquire()
     try {
       while (!finished && !isDone.get()) {
-        val nextEvent = Option(queue.poll())
+        val nextEvent = Option(queue.poll(1, TimeUnit.SECONDS))
```

Comment:
The 1 second time is arbitrary.

The loop should terminate when `isDone` is set, so probably shouldn't poll indefinitely. From reading the code, `isDone` ought to be set whenever there's an error creating global resources or whenever SBT calls `done` on the runner. I haven't been able to simulate either of these cases.

Reply:
I think 1 second is fine. 

It's really hard to test a test framework via the SBT interface, so as long as the build still builds, I think your changes are fine 👍

Reply:
I tested with scala-cli and SBT, and it seems to work fine.

You can reproduce the spinlock with 
```
scala-cli test https://gist.github.com/zainab-ali/867fbc095d2ab3ceda498b29bc2b8aba
```
Here's a screenshot of the CPU usage before:
![before](https://github.com/typelevel/weaver-test/assets/12057118/332f444b-8344-478c-a814-86beb2da5a49)

And after (with the current changes):
![after](https://github.com/typelevel/weaver-test/assets/12057118/3e2bfc83-398d-4663-83a4-ee0309036e81)

Reply:
Awesome, that's sufficient to get it merged then 👍